### DISCUSSION TITLE
Consider removing "Load X More Items"

### DISCUSSION THREAD

**Discussion by francislavoie** [community/community#37061] [Mobile]
When loading an issue with more than 10 or so items (which is like... all of them), we get a `LOAD X MORE ITEMS` button.

This is bad UX, because usually this causes important context for the highlighted item (e.g. from the notification) to be hidden by default, so it adds an extra step and friction to the experience.

On top of this, it often loses your scroll position after tapping it so it's jarring to get back to the same place.

If performance is a concern, virtualization should be used to avoid rendering/processing items that are off-screen.

For very long threads, having a "jump to the bottom" button near the bottom of the screen would be handy, once the button is removed and all items shown by default. Currently that's not needed of course because the total amount of items displayed is so low that it doesn't take much to scroll the the bottom.

Related to this is https://github.com/orgs/community/discussions/5004 which I think is important to make this feel better, by being able to refresh from the bottom of the screen to load in new replies.

Some discussion for context: https://twitter.com/_francislavoie/status/1584301901876977664

**Comment by aconsuegra** [comment]
Hi @francislavoie, thanks for your feedback! 

> On top of this, it often loses your scroll position after tapping it so it's jarring to get back to the same place.

This definitely sounds like a bug, could you tell me the platform you are using so I can create an issue for the team?

Regarding the UX issue of highlighting only the comment that created the notification, I will pass this feedback to our design team so they can re-evaluate the current approach and come up with a better one. Thanks again, and keep the feedback coming.

**Reply by francislavoie** [reply]
> This definitely sounds like a bug, could you tell me the platform you are using so I can create an issue for the team?

Android 13, Pixel 6 Pro.

I can't make it happen reliably, but I think I have a clue what's going on from trying it a bunch for 10 or so minutes.

When you tap on a notification (95% of my app usage is this) it highlights the first unread message via an anchor link. This tends to mean scrolling to the item and collapsing everything above it.

It seems that if the first time you tap the button to expand, if any of the comments that load in above had any "heavier" content (I'm thinking image embeds, code blocks, etc) then those cause the comments to resize, which pushes down the current position.

On subsequent attempts, it seems that content is already cached, so no push-down happens and the highlighted comment stays in place.

Either way though, there is a not-insignificant amount of flicker when loading in content, which adds to the feeling of jank.

If I open an issue without an anchor link, the app is much more generous about the amount of comments it'll load up-front, not collapsing as many above. Frankly, I think it would be best if anchor links didn't collapse anything above. Is it possible the collapsing was implemented as a workaround for the jank of additional content (images etc) loading in above after setting the scroll position, moving away from the highlighted comment? 🤔

Another annoyance I noticed was anchor links to PR approvals are awkward because it loads a screen where only the approval (and I assume the review comments attached) is visible. For example, use this link https://github.com/caddyserver/caddy-docker/pull/267#pullrequestreview-1153987956 it only shows the approval and nothing else at all until I hit back which then causes the rest of the PR to load in. I'm not sure what the best approach would be here but it is kinda annoying to have a separate screen that it loads into when coming from a notification. I'd prefer if it loaded the main view scrolled to the approval item instead.

**Reply by francislavoie** [reply]
What's more, I notice that if I hit back to go to the main view for something (e.g. PR approval, and this very thread) then it starts at the top of the main view, instead of being focused at the source item (anchor item). It should probably focus on the place I was just at so that I know in-context where I came from. I think that's probably a large reason for some feeling of jank as well.

**Reply by francislavoie** [reply]
If I tap this link https://github.com/community/community/discussions/37061#discussioncomment-3957147 from my notifications then go back (OS back action) it shows the top of the discussion instead of scrolling to the item I just came from.

If I tap on "view thread" from the main view though, then when I go back, it does retain the correct position (same as before).

Is it a situation of the "previous scroll position" being null if I came directly from a notification? Is it just missing a bit of code to calculate what that should be based on the current view before going back?

**Reply by francislavoie** [reply]
@aconsuegra does this sufficiently clarify the ask?

**Reply by aconsuegra** [reply]
Sure it does! Thanks a lot for your feedback.

We are evaluating how to revamp and improve the experience with deep links and content loading. Stay tuned.
